packages/ERTP/src/typeGuards.js

```diff
@@ -0,0 +1,51 @@
+import { assert } from '@agoric/assert';
+import { Nat } from '@agoric/nat';
+
+/**
+ * Type guard for SetValue
+ *
+ * @param {Value} value
+ * @returns {value is SetValue}
+ */
+export const isSetValue = value => Array.isArray(value);
```

Comment:
Either
```suggestion
export const isSetValue = value => {
  mustBeComparable(value);
  return Array.isArray(value);
};
```
or
```suggestion
export const isSetValue = value => isComparable(value) && Array.isArray(value);
```
depending on what you want this test to mean. I see from `isNatValue` below, which seems to be a sibling of this guard, that these guards may indicate failure by throwing or saying `false`. But when should it do which? What assurance does the programmer have if the guard returns false rather than throwing? From the other side, what invariant must the caller satisfy so that the guard returns rather than throwing? From this pair, I think only the second question has a clear answer: If the input is a valid `Value` then the guard will not throw.

Reply:
Above I did take a shortcut. I know that if a value both `isComparable(value)` and `isArray(value)`, then it is also `passStyleOf(value) === 'copyArray'`. So I used the cheaper array test rather than the clearer one. Feel free to use the clearer one instead.

Reply:
`isComparable` is a deep test. Both `isArray` and `passStyleOf(...) === 'copyArray'` are shallow tests. With the two together, we also know that all the elements of the array are `Comparable`, which, IIRC, is what we agreed is the defining criteria of being a valid SET value.

Reply:
I went back and looked at `setMathHelpers.doCoerce` and see that it is both more restrictive and less restrictive than what I wrote above. I forgot the `checkDupes` test above! We can't just make them the same though -- they are asked to do different jobs. A type guard is a validator. `doCoerce` is a coercer. But they should share more code and be more consistent with each other. Building the coercer on the validator would seem clean. In that case, the duplication checking should be moved into the validator.

Reply:
`mustBeComparable(...)` is like `assert(isComparable(...))` but with better error messages.